packages/itertools/_test.pony

```diff
@@ -455,6 +455,14 @@ class iso _TestIterTakeWhile is UnitTest
       Iter[I64](input.values())
         .take_while({(x) ? => error })
         .collect(Array[I64]))
+    
+    let infinite =
+      Iter[USize](Range(0, 3))
+        .cycle()
+        .take_while({(n) => n < 2 })
+    h.assert_eq[USize](0, infinite.next()?)
+    h.assert_eq[USize](1, infinite.next()?)
+    h.assert_false(infinite.has_next())
```

Comment:
Do you think it makes sense to verify that a call to `infinite.next()` is also properly erroring?

Reply:
This is a very good question and relates to a question that I was planning to ask in the next sync about what it means for an iterator to be "well behaved". I leave it out currently since some of the most common iterators will continue to give values after `has_next` returns false (such as Range).